### PR TITLE
Document `head_of_constr` as internal (in generalized rewriting chapter).

### LINKED ISSUE #16844: Tactic `head_of_constr` is not documented
It was introduced in fef54f7bae3342686e8005879ed89253ea19c3aa and it is used once in `theories/Classes/Morphisms.v` and `theories/Classes/CMorphisms.v`.

If it is useful, we should document it.

### PR DESCRIPTION
**Pull request by Zimmi48**
Closes #16844.

### CODE REVIEW

File: doc/sphinx/addendum/generalized-rewriting.rst [hunk 1]
```diff
@@ -659,6 +659,10 @@ new :tacn:`setoid_rewrite` tactic differs slightly from the old one and
 :tacn:`rewrite`.
 
 
+.. tacn:: head_of_constr @ident @one_term
+
+   Internal, will be removed without warning, do not use.
```
Comment (jfehrle) [inline doc/sphinx/addendum/generalized-rewriting.rst:664]:
```suggestion
   For internal use only.  It may be removed without warning.  Do not use.
```

**Review by jfehrle — COMMENTED** [review]
Why do you want to move the description from the ltac chapter to generalized rewriting?

**Comment by Zimmi48** [comment]
> Why do you want to move the description from the ltac chapter to generalized rewriting?

Because it's an internal tactic used in the generalized rewriting machinery and nowhere else, IIUC.

**Comment by Zimmi48** [comment]
Updated.

**Review by jfehrle — APPROVED** [review]
I would leave it where it was, but we can do it your way.

@coqbot: merge now

**Comment by jfehrle** [comment]
@coqbot: merge now

**Pull request merged**